### PR TITLE
Integration with the UPV mesos approach for bigsea

### PR DESCRIPTION
**Pull request by igornatanael**
This pull request contains the new plugin for spark progress monitoring, but considering some limitations with UPV infrastructure. The limitation consists that when running the broker from outside of the UPV, it is not possible to access the spark master node api to get the information about the tasks done because there is no public ip for those vms. 

This plugin is basically similar to the regular spark-progress plugin, but instead of using requests python lib to get information from spark master api during the execution, this plugin access via ssh a vm inside of the UPV network and from there execute a curl command and parse the response to a python-like format. The others commands, as calculating diff error, are pretty similar to the other.

### CODE REVIEW

File: monitor/plugins/spark_progress_upv.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                application_progress_error['value'] = error
+                application_progress_error['timestamp'] = time.time() * 1000
+                application_progress_error['dimensions'] = self.dimensions
+
+                print application_progress_error['value']
+
+                self.monasca.send_metrics([application_progress_error])
+
+            time.sleep(MONITORING_INTERVAL)
+
```
Comment (betobetodev) [inline monitor/plugins/spark_progress_upv.py:114]: White line.

File: monitor/plugins/spark_progress_upv.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        date_time = datetime.strptime(gmt_timestamp, '%Y-%m-%dT%H:%M:%S.%fGMT')
+        date_time = date_time.replace(tzinfo=pytz.utc).astimezone(local_tz)
+        date_time = date_time.replace(tzinfo=None)
+        datetime_now = datetime.now()
+        elapsed_time = datetime_now - date_time
+
+        return elapsed_time.seconds
+
+    def _discover_id_from_spark(self):
+        for i in range(30):
```
Comment (betobetodev) [inline monitor/plugins/spark_progress_upv.py:133]: Is 30 the number of attempts? If yes, create a generic parameter to this.

Comment (armstrongmsg) [reply monitor/plugins/spark_progress_upv.py:133]: I agree.

File: monitor/plugins/spark_progress_upv.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                print application_progress_error['value']
+
+                self.monasca.send_metrics([application_progress_error])
+
+            time.sleep(MONITORING_INTERVAL)
+
+
+    def _get_elapsed_time(self, gmt_timestamp):
+        try:
+            local_tz = tzlocal.get_localzone()
```
Comment (armstrongmsg) [inline monitor/plugins/spark_progress_upv.py:118]: Is this working now?

**Review by armstrongmsg — CHANGES_REQUESTED** [review]

File: monitor/plugins/spark_progress_upv.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
+
+                self.remaining_time = self.remaining_time - elapsed_time
+
+                self.current_job_id = len(jobs) - 1
+
+                # Job Time
+                self.job_expected_time = (self.remaining_time
+                                 / (float(self.number_of_jobs)
```
Comment (tellesnobrega) [inline monitor/plugins/spark_progress_upv.py:84]: I don't understand this line break. Prefer not to use \ .

File: monitor/plugins/spark_progress_upv.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                                 - float(self.current_job_id)))
+
+            elif current_job['status'] == 'RUNNING':
+                # Job Progress
+                job_progress = (current_job['numCompletedTasks']
+                                / float(current_job['numTasks']))
```
Comment (tellesnobrega) [inline monitor/plugins/spark_progress_upv.py:90]: I don't really understand this line break.

File: monitor/plugins/spark_progress_upv.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+                # Elapsed Time
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
```
Comment (tellesnobrega) [inline monitor/plugins/spark_progress_upv.py:94]: Indentation  again, but only needed if new patch is being sent

**Pull request merged**